### PR TITLE
Start express when tests run

### PR DESCRIPTION
**Pull request by nodanaonlyzuul**
@nonword / @jobinthomasnypl - developers currently have to start the app, then jump to another tab to run `npm test`. After this PR - running `npm test` will start up express.

It will start on 3000 - so you can't have your dev envrionment up on 3000 but I'll
follow up with a PR to make the test port configurable.

### test_helper.js 

This branch creates  `./test/test_helper.js`
All tests require it and it reduces boiler plate by:

* Setting `process.env.NODE_ENV = 'test'`.
* requiring 'chai' once and exposing it as a global for our tests.
* Starting up the server

### Reduce Boilerplate in Tests

* https://github.com/NYPL-discovery/discovery-api/commit/238191b1a7fd0de763e6ab8598d74a9fbd20ab33: Tell standard once, and only once, that our tests will have mocha syntax that makes use of a global named `expect()`. This cuts down on the [magic comments](http://eslint.org/docs/user-guide/configuring) we need to put in tests.

### Stop Logging To Console In Test Mode

* https://github.com/NYPL-discovery/discovery-api/commit/4e159816593d48f3d05b4a72d27e73e87635e5d4: Now that npm test starts it's own server...we don't want log cluttering out terminals. This is an entry to future NODE_ENV specific logging options [we want to do](https://www.youtube.com/watch?v=dQw4w9WgXcQ).

### CODE REVIEW

File: test/test_helper.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+process.env.NODE_ENV = 'test'
+require('../app.js')
```
Comment (nonword) [inline test/test_helper.js]: `express.listen` seems to fail silently if the port is already attached, which means it's possible the test suite will be run against some other app attached to 3000. A global `before` in test_helper.js could quickly check for this situation like this:

```js
// Make sure nothing else bound to port:
before(function () {
  return require('request').get('http://localhost:' + config.get('port'), function (err, response, body) {
    if (err && err.code === 'ECONNREFUSED') {
      // No test server is running, so safe to instantiate one from local code:
      require('../app.js')
    } else {
      console.err(`Something else already listening on ${config.get('port')}. Aborting.`)
      process.exit()
    }   
  })  
})```

File: test/scsb_rest_client.test.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-/* eslint-env mocha */
 /* eslint no-new:0*/
-const expect = require('chai').expect
-const SCSBRestClient = require('../lib/scsb_rest_client.js')
+
+require('./test_helper')
```
Comment (jobinthomasnypl) [inline test/scsb_rest_client.test.js]: I tried @nonword's suggestion. True, I didn't know that either

File: test/test_helper.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+require('../lib/globals')
+global.expect = require('chai').expect
```
Comment (jobinthomasnypl) [inline test/test_helper.js]: This is great that expect is in one place and everyone else can get it from this helper file. Only issue I guess is standard is going to complain saying expect is not defined.

File: test/test_helper.js (hunk 1 above)
Comment (nodanaonlyzuul) [inline test/test_helper.js]: @nonword & spoke: We're not sure what was going on with his localhost earlier - but we see that running tests with a dev env up - loudly will throw a `listen EADDRINUSE :::3000` error.

We're going to merge in this PR - and follow up with a ticket to configure another port for tests.

**Pull request merged**